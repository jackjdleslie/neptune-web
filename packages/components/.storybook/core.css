:root {
  --primary-color: rgb(0, 185, 255);
  --pay-color: rgb(46, 208, 110);
  --danger-color: #f53636;
  --text-on-primary: white;

  --font: Averta, 'Avenir W02', Avenir, Helvetica, Arial, sans-serif;
  --weight-1: 400;
  --weight-2: 600;
  --weight-3: 800;

  --space-1: 8px;
  --space-2: 16px;
  --space-3: 24px;
  --space-4: 32px;
  --space-5: 40px;
  --space-6: 48px;

  --border-radius: 3px;
}

.primary-background {
  background: var(--primary-color);
}
