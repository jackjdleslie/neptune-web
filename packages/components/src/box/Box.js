/** @jsx jsx */
/* eslint-disable */
import { jsx } from '@emotion/core';
import Types from 'prop-types';

import { Size, JustifyContent, AlignItems, Breakpoint, Spacer } from '../common';

const mediaQueries = Object.values(Breakpoint).map(bp => `@media (min-width: ${bp}px)`);

const Box = props => {
  const {
    as: Element,
    size,
    justifyContent,
    alignItems,
    children,
    className,
    marginX,
    paddingX,
    marginY,
    paddingY,
    customMediaQueries,
  } = props;

  const getFlex = breakpoint => {
    const style = { flex: 0, display: 'none' };
    if (size && size[breakpoint]) {
      style.flex =
        size[breakpoint] <= 1 ? `0 1 ${size[breakpoint] * 100}%` : `0 0 ${size[breakpoint]}px`;
      style.display = 'flex';
    }
    return style;
  };

  const style = {
    [customMediaQueries[0]]: {
      ...getFlex(Size.EXTRA_SMALL),
    },
    [customMediaQueries[1]]: {
      ...getFlex(Size.SMALL),
    },
    [customMediaQueries[2]]: {
      ...getFlex(Size.MEDIUM),
    },
    [customMediaQueries[3]]: {
      ...getFlex(Size.LARGE),
    },
    [customMediaQueries[4]]: {
      ...getFlex(Size.EXTRA_LARGE),
    },
  };

  if (justifyContent) {
    style.justifyContent = justifyContent;
  }
  if (alignItems) {
    style.alignItems = alignItems;
  }

  style.margin = `${Spacer * marginY}px  ${Spacer * marginX}px`;
  style.padding = `${Spacer * paddingY}px  ${Spacer * paddingX}px`;

  return size ? (
    <Element className={className} css={style}>
      {children}
    </Element>
  ) : null;
};

Box.propTypes = {
  alignItems: Types.oneOf([...Object.values(AlignItems)]),
  children: Types.oneOfType([Types.arrayOf(Types.node), Types.node]),
  className: Types.string,
  justifyContent: Types.oneOf([...Object.values(JustifyContent)]),
  marginX: Types.number,
  marginY: Types.number,
  paddingX: Types.number,
  paddingY: Types.number,
  as: Types.elementType,
  size: Types.shape({
    lg: Types.number,
    md: Types.number,
    sm: Types.number,
    xl: Types.number,
    xs: Types.number,
  }).isRequired,
  customMediaQueries: Types.arrayOf(Types.string),
};

Box.defaultProps = {
  justifyContent: null,
  alignItems: null,
  children: null,
  className: '',
  paddingX: 0,
  marginX: 0,
  paddingY: 0,
  marginY: 0,
  as: 'div',
  customMediaQueries: mediaQueries,
};

export default Box;
