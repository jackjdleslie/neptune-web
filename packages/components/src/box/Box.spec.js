import serializer from 'jest-emotion';
/** @jsx jsx */
import { jsx } from '@emotion/core';
import renderer from 'react-test-renderer'; // eslint-disable-line import/no-extraneous-dependencies

import Box from './';

const Width = { xs: 1, sm: 1 / 2, md: 30, lg: 40, xl: 0 };
const customMediaQueries = [0, 10, 20, 30, 40].map(bp => `@media (min-width: ${bp}px)`);

expect.addSnapshotSerializer(serializer);

const props = {
  tagHtml: 'section',
  size: Width,
  marginX: 1,
  marginY: 2,
  paddingX: 3,
  paddingY: 4,
  justifyContent: 'center',
  alignItems: 'end',
  extraClassNames: ['anotherClass1', 'anotherClass2'],
};

describe('Box', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Box {...props}>
        <div>This is a children.</div>
      </Box>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('renders correctly with custom media queries', () => {
    const component = renderer.create(
      <Box {...props} customMediaQueries={customMediaQueries}>
        <div>This is a children.</div>
      </Box>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
