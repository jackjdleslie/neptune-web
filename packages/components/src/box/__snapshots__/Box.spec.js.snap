// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
.emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  margin: 16px 8px;
  padding: 32px 24px;
}

@media (min-width:0px) {
  .emotion-0 {
    -webkit-flex: 0 1 100%;
    -ms-flex: 0 1 100%;
    flex: 0 1 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:576px) {
  .emotion-0 {
    -webkit-flex: 0 1 50%;
    -ms-flex: 0 1 50%;
    flex: 0 1 50%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .emotion-0 {
    -webkit-flex: 0 0 30px;
    -ms-flex: 0 0 30px;
    flex: 0 0 30px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:992px) {
  .emotion-0 {
    -webkit-flex: 0 0 40px;
    -ms-flex: 0 0 40px;
    flex: 0 0 40px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
    display: none;
  }
}

<section
  className="anotherClass1 anotherClass2 emotion-0"
>
  <div>
    This is a children.
  </div>
</section>
`;

exports[`Box renders correctly with custom media queries 1`] = `
.emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  margin: 16px 8px;
  padding: 32px 24px;
}

@media (min-width:0px) {
  .emotion-0 {
    -webkit-flex: 0 1 100%;
    -ms-flex: 0 1 100%;
    flex: 0 1 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:10px) {
  .emotion-0 {
    -webkit-flex: 0 1 50%;
    -ms-flex: 0 1 50%;
    flex: 0 1 50%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:20px) {
  .emotion-0 {
    -webkit-flex: 0 0 30px;
    -ms-flex: 0 0 30px;
    flex: 0 0 30px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:30px) {
  .emotion-0 {
    -webkit-flex: 0 0 40px;
    -ms-flex: 0 0 40px;
    flex: 0 0 40px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:40px) {
  .emotion-0 {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
    display: none;
  }
}

<section
  className="anotherClass1 anotherClass2 emotion-0"
>
  <div>
    This is a children.
  </div>
</section>
`;
